File: src/main/scala/ReadData.scala
package main.scala

import org.apache.spark.sql.SparkSession


object ReadData {
    def main(args: Array[String]): Unit = {
        val spark = SparkSession
            .builder
            .appName("read data folder")
            .master("local[*]")
            .getOrCreate()

        val data = spark.read
            .format("csv")
            .option("inferSchema", "true")
            .option("header", "true")
            .load("./src/main/scala/data/mnm_dataset_output")

        data.show(5, false)

        spark.stop()
    }
}

File: src/read_data.py
import argparse
from pyspark.sql import SparkSession


def main(data_path: str) -> None:
    spark = SparkSession.builder.appName("read data with spark").master("local[*]").getOrCreate()

    data_path = "./data/mnm_dataset.csv"
    data = spark.read.format("csv").option("header", "true").option("inferSchema", "true").load(data_path)
    data.show(5, False)

    spark.stop()

    return None


if __name__ == "__main__":
    parser = argparse.ArgumentParser("read data source path")
    parser.add_argument("--data_path", required=True, help="path to data sources like csv files")
    args = parser.parse_args()

    data_path = args.data_path

    main(data_path)
